flow/Knobs.h

```diff
@@ -43,22 +43,30 @@ enum class ConfigDBType {
 	PAXOS,
 };
 
+enum class Atomic { YES, NO };
```

Comment:
This is turning out to be some amount of trouble, as it requires `include`ing `flow/BooleanParam.h` in `flow/Knobs.h`. This then causes a bunch of compile errors like

```
In file included from /mnt/ephemeral/ljoswiak/foundationdb/fdbrpc/AsyncFileNonDurable.actor.cpp:21:
In file included from /mnt/ephemeral/ljoswiak/foundationdb/fdbrpc/AsyncFileNonDurable.actor.h:27:
In file included from /mnt/ephemeral/ljoswiak/foundationdb/fdbrpc/AsyncFileNonDurable.actor.h:31:
In file included from /mnt/ephemeral/ljoswiak/foundationdb/flow/flow.h:23:
In file included from /mnt/ephemeral/ljoswiak/foundationdb/flow/Arena.h:27:
In file included from /mnt/ephemeral/ljoswiak/foundationdb/flow/FastAlloc.h:25:
In file included from /mnt/ephemeral/ljoswiak/foundationdb/flow/Error.h:30:
In file included from /mnt/ephemeral/ljoswiak/foundationdb/flow/Knobs.h:27:
In file included from /mnt/ephemeral/ljoswiak/foundationdb/flow/BooleanParam.h:23:
/mnt/ephemeral/ljoswiak/foundationdb/flow/Trace.h:205:3: error: use of undeclared identifier 'UNSTOPPABLE_ASSERT'
                UNSTOPPABLE_ASSERT(false);
                ^
/mnt/ephemeral/ljoswiak/foundationdb/flow/Trace.h:494:8: error: field has incomplete type 'Error'
        Error err;
              ^
/mnt/ephemeral/ljoswiak/foundationdb/flow/Trace.h:380:32: note: forward declaration of 'Error'
        TraceEvent& error(const class Error& e, bool includeCancelled = false) {
                                      ^
/mnt/ephemeral/ljoswiak/foundationdb/flow/Trace.h:554:5: error: function-like macro 'CENABLED' is not defined
#if CENABLED(0, NOT_IN_CLEAN)
```

presumably because `BooleanParam.h` is including `Trace.h`.

Reply:
Oh, I see. Yeah, using your own `enum class` here makes sense.